scipy/stats/tests/test_fit.py

```diff
@@ -159,3 +162,279 @@ def test_nnlf_and_related_methods(dist, params):
+            stats.fit(self.dist, self.data, self.shape_bounds_a,
+                      loc_bounds=loc_bounds)
+
+    def test_scale_bounds_iv(self):
+
+        message = "`binom` is an instance of `rv_discrete`..."
+        scale_bounds = (1, 10)
+        with pytest.warns(RuntimeWarning, match=message):
+            stats.fit(self.dist, self.data, self.shape_bounds_a,
+                      scale_bounds=scale_bounds)
+
+        message = "`scale_bounds` must be a tuple containing exactly two..."
+        scale_bounds = (0, 1, 2)
+        with pytest.raises(ValueError, match=message):
+            stats.fit(stats.norm, self.data, scale_bounds=scale_bounds)
+        with pytest.raises(ValueError, match=message):
+            stats.fit(stats.norm, self.data, [], scale_bounds=scale_bounds)
+
+        message = "There are no values for `scale` on the interval..."
+        scale_bounds = (10, 0)
+        with pytest.raises(ValueError, match=message):
+            stats.fit(stats.norm, self.data, scale_bounds=scale_bounds)
+
+        message = "The intersection of user-provided bounds for `scale`"
+        scale_bounds = (1, np.inf)
+        with pytest.raises(ValueError, match=message):
+            stats.fit(stats.norm, self.data, scale_bounds=scale_bounds)
+
+    # these distributions are tested separately
+    skip_basic_fit = {'nhypergeom', 'boltzmann', 'nbinom',
+                      'randint', 'yulesimon', 'nchypergeom_fisher',
+                      'nchypergeom_wallenius'}
+
+    @pytest.mark.parametrize("dist_name", dict(distdiscrete))  # type: ignore[arg-type]  # noqa
+    def test_basic_fit(self, dist_name):
+        if dist_name in self.skip_basic_fit or not isinstance(dist_name, str):
+            pytest.skip("Tested separately.")
+
+        N = 5000
+        dist_data = dict(distcont + distdiscrete)
+        rng = np.random.default_rng(self.seed)
+        dist = getattr(stats, dist_name)
+        shapes = dist_data[dist_name]
+        shape_bounds = np.array([shapes, shapes], dtype=np.float64).T
+        shape_bounds[:, 0] /= 10  # essentially all shapes are > 0
+        shape_bounds[:, 1] *= 10
+        loc_bounds = (0, 10)
+        scale_bounds = (0, 10)
+        loc = rng.uniform(*loc_bounds)
+        scale = rng.uniform(*scale_bounds)
+
+        if getattr(dist, 'pmf', False):
+            loc = np.floor(loc)
+            data = dist.rvs(*shapes, size=N, loc=loc, random_state=rng)
+            res = stats.fit(dist, data, shape_bounds=shape_bounds,
+                            loc_bounds=loc_bounds, optimizer=self.opt)
+            ref = list(shapes) + [loc]
+        if getattr(dist, 'pdf', False):
+            data = dist.rvs(*shapes, size=N, loc=loc, scale=scale,
+                            random_state=rng)
+            res = stats.fit(dist, data, shape_bounds=shape_bounds,
+                            loc_bounds=loc_bounds,
+                            scale_bounds=scale_bounds, optimizer=self.opt)
```

Comment:
Do we want to add tests to only optimize loc or/and scale? We will for sure have user not specifying `shape_bounds` but the others.

Reply:
I don't think there are any continuous distributions that don't require `shape_bounds`, and discrete distributions don't use `scale`, so it's not possible to have a test in which we pass only `scale_bounds` but no `shape_bounds`. We could pass in only `loc_bounds` but no `shape_bounds` for a few distributions, like `bernoulli`. Is that what you mean? 

We have plenty of tests that fit all three, some that fit shape and loc, and some that fit only the shape. I guess another combination we could add is passing `shape_bounds` and `scale_bounds` without `loc_bounds`, if you'd like to see that.